### DISCUSSION TITLE
kanagawa colour scheme

### DISCUSSION THREAD

**Discussion by LaughingBubba** [nix-community/nixvim#443] [Q&A] [answered]
Hi I'm trying to figure out how setup [kanagawa colour scheme](https://github.com/rebelot/kanagawa.nvim) in `nixvim`.

There is a nix package for it: https://search.nixos.org/packages?channel=23.05&show=vimPlugins.kanagawa-nvim&from=0&size=50&sort=relevance&type=packages&query=kanagawa 

I got `gruvbox` scheme working with a one liner in the `default.nix` file but `kanagawa` is eluding me.

`flake.nix`
```nix
{
  description = "A nixvim configuration";

  inputs = {
    nixvim.url = "github:pta2002/nixvim";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = {
    nixpkgs,
    nixvim,
    flake-utils,
    ...
  } @ inputs: let
    config = import ./config; # import the module directly
  in
    flake-utils.lib.eachDefaultSystem (system: let
      nixvimLib = nixvim.lib.${system};
      pkgs = import nixpkgs {inherit system;};
      nixvim' = nixvim.legacyPackages.${system};
      nvim = nixvim'.makeNixvimWithModule {
        inherit pkgs;
        module = config;
      };
    in {
      checks = {
        # Run `nix flake check .` to verify that your config is not broken
        default = nixvimLib.check.mkTestDerivationFromNvim {
          inherit nvim;
          name = "A nixvim configuration";
        };
      };
     ### ADDED 
      programs.nixvim = {
        extraPlugins = [ pkgs.vimPlugins.kanagawa-nvim ];
        colorscheme = "kanagawa-nvim";
      };
     ###
      packages = {
        # Lets you run `nix run .` to start nixvim
        default = nvim;
      };
    });
}
```
`./config/default.nix`
```nix
{
  # Import all your configuration modules here
  imports = [
    ./bufferline.nix
  ];
  #colorschemes.gruvbox.enable = true;

 ### Doesn't recognise scheme
  #colorschemes.kanagawa-nvim.enable = true;
}
```

**Comment by GaetanLepage** [comment]
Hi !
This is normal. Indeed, nixvim supports a few colorschemes for which we have wrapped the options.
It is not because one colorscheme plugin (or any vim plugin in general) is packages in nixpkgs that it is wrapped in nixvim.
You can, however, add any plugin packaged in `nixpkgs` in the `nixvim.extraPlugins` list.

I made a [PR](https://github.com/pta2002/nixvim/pull/444) to add support for kanagawa though :)

**Comment by siph** [comment]
To expand a little. This would work if you put it in the config, and you're also calling the colorscheme by the wrong name.

You can take the added part out of `flake.nix` and `./config/default.nix` would look like:
```nix
{ pkgs, ... }:
{
  imports = [
    ./bufferline.nix
  ];
  extraPlugins = with pkgs; [ vimPlugins.kanagawa-nvim ];
  colorscheme = "kanagawa";
}

```

**Reply by GaetanLepage** [reply]
Thank you @siph for adding this precision.
The PR should be merged soon, so once merged, you will be able to simply set `colorschemes.kanagawa.enable = true` (and configure it if you wish).

**Reply by LaughingBubba** [reply]
Thanks for the prompt response. I cycled through a number of variations before posting which included "kanagawa" as the colorscheme. If figured I was missing something because `programs` wasn't declared anywhere and yet the `README.md` had `programs.nixvim` 

The suggested code works just dandy thanks and thanks @GaetanLepage for the PR as well.

**Reply by siph** [reply]
If you're making a standalone flake you won't need to use `programs`, that's for using it as a module in `nixos` or `home-manager`

**Reply by GaetanLepage** [reply, marked as answer]
@LaughingBubba the PR has just been merged ! If you are satisfied with the solution, could you please mark this thread as solved ?
